Automattic/simplenote-ios
# Add Sentry Support

Adds support for using Sentry via Tracks.

Most of the logic is over in https://github.com/Automattic/Automattic-Tracks-iOS/pull/103. Once that's merged, I'll update this PR with a real version number, instead of a `pod` branch.

**To Test**

1. First off, we'll need to make a couple of small changes to enable testing this:
- Comment out the `BuildConfiguration.is(.debug)` line in [SPPrivacyViewController.swift](https://github.com/Automattic/simplenote-ios/pull/304/files#diff-0893f721a13d096d145c51f82fe2c2b7R19). We want the test crash code to be available no matter what for our testing.
- Change the app's build configuration to Release (edit Scheme > Run > Build Configuration => Release)
- Run `bundle exec fastlane run configure_apply` in the root of the project in order to download the proper configuration.
2. Run the app on-device.
3. Disconnect the device from the host machine, and force-quit the app.
3. Relaunch the app and Log in, ensuring that your user is set up to send crash reports.
4. Navigate to Settings > Privacy Settings > Send a Test Crash (the app will crash)
4. Re-run the app, and let it idle for a moment (this'll ensure the crash report goes to Sentry)
5. Note your shiny new crash event in Sentry – you can see it under "Events" if you filter by `simplenote-ios`.
